Cluster creation panics due to IAM profile being nil

/kind bug

**What steps did you take and what happened:**

Ran `clusterctl create cluster ...` and was expecting to see a new cluster created from the latest master. It seems that in some case (e.g. bastion host) the IAM profile isn't available and the ec2.Instance struct has a nil field.

```
I0311 21:41:33.551443       1 plugins.go:39] Registered cluster provisioner "aws"
I0311 21:41:33.552720       1 reflector.go:131] Starting reflector *v1alpha1.Cluster (10h0m0s) from pkg/cache/internal/informers_map.go:126
I0311 21:41:33.552758       1 reflector.go:169] Listing and watching *v1alpha1.Cluster from pkg/cache/internal/informers_map.go:126
I0311 21:41:33.553038       1 reflector.go:131] Starting reflector *v1alpha1.Machine (10h0m0s) from pkg/cache/internal/informers_map.go:126
I0311 21:41:33.553064       1 reflector.go:169] Listing and watching *v1alpha1.Machine from pkg/cache/internal/informers_map.go:126
I0311 21:41:33.752543       1 controller.go:113] Reconciling Machine "controlplane-0"
I0311 21:41:33.752599       1 actuator.go:375] Checking if machine controlplane-0 for cluster test1 exists
I0311 21:41:33.752551       1 controller.go:89] Running reconcile Cluster for test1
I0311 21:41:33.752789       1 controller.go:127] reconciling cluster object test1 triggers idempotent reconcile.
I0311 21:41:33.752793       1 actuator.go:57] Reconciling cluster test1
I0311 21:41:33.754068       1 certificates.go:72] Reconciling certificates for cluster "test1"
I0311 21:41:33.754095       1 network.go:25] Reconciling network for cluster "test1"
I0311 21:41:33.754099       1 vpc.go:40] Reconciling VPC
I0311 21:41:33.759992       1 controller.go:213] Reconciling machine object controlplane-0 triggers idempotent create.
I0311 21:41:33.760019       1 actuator.go:127] Creating machine controlplane-0 for cluster test1
I0311 21:41:33.847476       1 instances.go:40] Looking for existing instance for machine "controlplane-0" in cluster "test1"
I0311 21:41:34.694759       1 actuator.go:115] Machine "controlplane-0" should join the controlplane: false
I0311 21:41:34.764797       1 vpc.go:71] Working on managed VPC "vpc-04c16223c37c430e6"
I0311 21:41:34.764899       1 subnets.go:40] Reconciling subnets
I0311 21:41:35.098565       1 subnets.go:135] Subnets available: [id=subnet-0b745be0eddedf850/az=us-west-2a/public=true id=subnet-07016e754187e9f22/az=us-west-2a/public=false]
I0311 21:41:35.098634       1 gateways.go:39] Reconciling internet gateways
I0311 21:41:35.189020       1 natgateways.go:40] Reconciling NAT gateways
I0311 21:41:35.327845       1 routetables.go:42] Reconciling routing tables
I0311 21:41:35.367886       1 routetables.go:51] Subnet "subnet-0b745be0eddedf850" is already associated with route table "rtb-097b77f603d94868c"
I0311 21:41:35.367957       1 routetables.go:51] Subnet "subnet-07016e754187e9f22" is already associated with route table "rtb-0823a33c9a6ff09d7"
I0311 21:41:35.367971       1 securitygroups.go:35] Reconciling security groups
I0311 21:41:35.529764       1 network.go:57] Reconcile network completed successfully
I0311 21:41:35.529780       1 bastion.go:46] Reconciling bastion host
E0311 21:41:35.852641       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
bazel-out/darwin-fastbuild/bin/external/io_bazel_rules_go/linux_amd64_pure_stripped/stdlib%/src/runtime/asm_amd64.s:522
GOROOT/src/runtime/panic.go:513
GOROOT/src/runtime/panic.go:82
GOROOT/src/runtime/signal_unix.go:390
pkg/cloud/aws/converters/instance.go:44
pkg/cloud/aws/services/ec2/bastion.go:121
pkg/cloud/aws/services/ec2/bastion.go:59
pkg/cloud/aws/actuators/cluster/actuator.go:79
vendor/sigs.k8s.io/cluster-api/pkg/controller/cluster/controller.go:128
vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
bazel-out/darwin-fastbuild/bin/external/io_bazel_rules_go/linux_amd64_pure_stripped/stdlib%/src/runtime/asm_amd64.s:1333
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe5da0e]

goroutine 175 [running]:
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
panic(0x15b7540, 0x293f8a0)
	GOROOT/src/runtime/panic.go:513 +0x1b9
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/converters.SDKToInstance(0xc0005ba000, 0xc0000b5130)
	pkg/cloud/aws/converters/instance.go:44 +0x21e
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/ec2.(*Service).describeBastionInstance(0xc000684140, 0xc00017a840, 0x4, 0xc00000dbc8)
	pkg/cloud/aws/services/ec2/bastion.go:121 +0x1f7
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/ec2.(*Service).ReconcileBastion(0xc000684140, 0x0, 0x0)
	pkg/cloud/aws/services/ec2/bastion.go:59 +0x170
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/cluster.(*Actuator).Reconcile(0xc0002244c0, 0xc0000ba000, 0x0, 0x0)
	pkg/cloud/aws/actuators/cluster/actuator.go:79 +0x2b0
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/cluster-api/pkg/controller/cluster.(*ReconcileCluster).Reconcile(0xc000014150, 0xc000720ea0, 0x7, 0xc000720e90, 0x5, 0x2954180, 0xc00003c240, 0xc000478540, 0x4057bb)
	vendor/sigs.k8s.io/cluster-api/pkg/controller/cluster/controller.go:128 +0x37c
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000430e60, 0x0)
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x18f
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00018c0a0)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00018c0a0, 0x3b9aca00, 0x0, 0x1, 0xc00023aba0)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbe
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00018c0a0, 0x3b9aca00, 0xc00023aba0)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x32a
```


**What did you expect to happen:**
Cluster created successfully.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

--- a/pkg/cloud/aws/converters/instance.go
+++ b/pkg/cloud/aws/converters/instance.go
@@ -41,12 +41,10 @@ func SDKToInstance(v *ec2.Instance) *v1alpha1.Instance {
 	// Extract IAM Instance Profile name from ARN
 	// TODO: Handle this comparison more safely, perhaps by querying IAM for the
 	// instance profile ARN and comparing to the ARN returned by EC2
-	if v.IamInstanceProfile.Arn != nil {
+	if v.IamInstanceProfile != nil && v.IamInstanceProfile.Arn != nil {
 		split := strings.Split(aws.StringValue(v.IamInstanceProfile.Arn), "instance-profile/")
-		if len(split) > 1 {
-			if split[1] != "" {
-				i.IAMProfile = split[1]
-			}
+		if len(split) > 1 && split[1] != "" {
+			i.IAMProfile = split[1]
 		}
 	}
 

